src/json/json_schema.rs

```diff
@@ -136,8 +136,22 @@ impl JsonSchema {
             }
             JsonSchema::Enum(a) => {
                 if let JsonSchema::Enum(b) = other {
-                    if a.ne(b) {
-                        return Valid::fail("expected Enum type".to_string()).trace(name);
+                    if a.len() != b.len() {
+                        return Valid::fail("expected proper Enum type".to_string()).trace(name);
+                    } else {
+                        let keys: HashSet<_> = a.keys().collect();
+                        for key in keys {
+                            let mut small = &b[key];
+                            let mut large = &a[key];
+                            if a[key].len() < b[key].len() {
+                                small = &a[key];
+                                large = &b[key];
+                            }
```

Comment:
I don't think it's good idea to hardcode it this way.. we should check if alias is really allowed

Reply:
I think allow alias option is just for reading proto files with aliases.
If you read proto file with aliases when not allowing them, then we are getting validation error.
If alias is allowed then we can have variants with name number for the example proto provided in the issue.

Reply:
> I don't think it's good idea to concat array it this way.. we should check if alias is really allowed.

Where is concatenation?

Reply:
Actually, we should not do small and large this way, if we are certain that a is from schema file and b is from proto file.

Reply:
> I think allow alias option is just for reading proto files with aliases. If you read proto file with aliases when not allowing them, then we are getting validation error. If alias is allowed then we can have variants with name number for the example proto provided in the issue. We use them this way to validate enum.

you are putting index as key which is incorrect. Yes you can keep value from proto as key but that’s not what it’s doing currently

Reply:
I am using value.number() as key for proto
And index of variants as key for enum from graphql schema. Which I am treating it in order defined by the user. If this is why you are talking about? Please clarify.
User can specify the number for each enum variant explicitly in graphql. Otherwise implicitly we do as above.
I think this clarifies.

Reply:
I am using value.number() as key for proto
And index of variants as key for enum from graphql schema. Which I am treating it in order defined by the user. If this is why you are talking about? Please clarify.
User can specify the number for each enum variant explicitly in graphql. Otherwise implicitly we do as above.
Let know if I am understanding it wrong.

Reply:
I am confused on how are you planning to check if alias is turned on? If it's not explicitly defined, we should throw an error

Reply:
I am not checking it explicitly.
If there are aliases, to read the proto correctly without validation error one should define option there in enum to allow alias.

I am not sure where that validation is performed in code. But I have tested an example for it.

If no alias then it will as usual pass through.
In this case the value: hashset in hashmap is of size one.
In this case Is Subset validation is between hashsets of same same size.
Otherwise if there are alias it's unequal size.

We should loop over keys of small hashset ie. For graphql schema enum.

Reply:
In will correct it now.

Reply:
you need to check for `allow_alias`

Reply:
The code must be strict. We need to throw an error if alias is false/undefined.

example: imagine there are 3 fields defined in an enum in proto file but the user forgot and defined just 3 fields in graphql config, in that case we must throw an error else the user will just get "2" as response instead of actual enum field name, which is quite incorrect.

Reply:
I think the alias property in proto files is handled by proto reader.

Reply:
let me try with an example.

Reply:
> I think the alias property in proto files is handled by proto reader.

it's not handled anywhere, we are dropping all that information as soon as we read proto file 

check my comment: https://github.com/tailcallhq/tailcall/issues/1626#issuecomment-2032455118

we do not preserve any kind of options in our descriptor, which needs a fix.

Reply:
sorry not by proto reader, but someone.
not handling allow_alias property explicitly.
but its is checking like this:
ERROR Invalid Configuration
Caused by:
  • enum number '0' has already been used [at Query.hello.@grpc]

Reply:
yes, we need to check if allow_alias is set to true. If it's true then we should not throw the error else we must.

Reply:
checking initially at proto reading stage is good.
But i didnt understand why this error wont come if set to true.
ERROR Invalid Configuration
Caused by:
• enum number '0' has already been used [at Query.hello.https://github.com/grpc]